lib/os/fdtable.c

```diff
@@ -211,17 +268,46 @@ off_t lseek(int fd, off_t offset, int whence)
 }
 FUNC_ALIAS(lseek, _lseek, off_t);
 
-int ioctl(int fd, unsigned long request, ...)
+int z_impl_sys_ioctl(int fd, unsigned long request, va_list args)
 {
-	va_list args;
-	int res;
-
 	if (_check_fd(fd) < 0) {
 		return -1;
 	}
 
+	return fdtable[fd].vtable->ioctl(fdtable[fd].obj, request, args);
+}
+
+#ifdef CONFIG_USERSPACE
+ssize_t z_vrfy_sys_ioctl(int fd, unsigned long request, va_list args)
+{
+/* This doesn't work so far, until we start to fish inside a particular
+ * va_list structure for a particular arch. Maybe later.
```

Comment:
"maybe later"

If you are going to send PRs with obvious security problems like this please consider sending them as a draft.